elixir/apps/api/test/api/controllers/gateway_group_controller_test.exs

```diff
@@ -0,0 +1,202 @@
+defmodule API.GatewayGroupControllerTest do
+  use API.ConnCase, async: true
+  alias Domain.Gateways.Group
+
+  setup do
+    account = Fixtures.Accounts.create_account()
+    actor = Fixtures.Actors.create_actor(type: :api_client, account: account)
+
+    %{
+      account: account,
+      actor: actor
+    }
+  end
+
+  describe "index" do
+    test "returns error when not authorized", %{conn: conn} do
+      conn = post(conn, "/v1/gateway_groups")
+      assert json_response(conn, 401) == %{"errors" => %{"detail" => "Unauthorized"}}
+    end
+
+    test "lists all gateway groups", %{conn: conn, account: account, actor: actor} do
+      gateway_groups = for _ <- 1..3, do: Fixtures.Gateways.create_group(%{account: account})
+
+      conn =
+        conn
+        |> authorize_conn(actor)
+        |> put_req_header("content-type", "application/json")
+        |> get("/v1/gateway_groups")
+
+      assert %{
+               "data" => data,
+               "metadata" => %{
+                 "count" => count,
+                 "limit" => limit,
+                 "next_page" => next_page,
+                 "prev_page" => prev_page
+               }
+             } = json_response(conn, 200)
+
+      assert count == 3
+      assert limit == 50
+      assert is_nil(next_page)
+      assert is_nil(prev_page)
+
+      data_ids = Enum.map(data, & &1["id"]) |> MapSet.new()
+      gateway_group_ids = Enum.map(gateway_groups, & &1.id) |> MapSet.new()
+
+      assert MapSet.equal?(data_ids, gateway_group_ids)
```

Comment:
What if we add a helper to `ConnCase` called `equal_ids?(list, list)` which will do all of this? This code is duplicated in a lot of places.

Reply:
Done